cmake/FindImath.cmake

```diff
@@ -0,0 +1,121 @@
+# Copyright Contributors to the OpenVDB Project
+# SPDX-License-Identifier: MPL-2.0
+#
+#[=======================================================================[.rst:
+
+FindImath
+-----------
+
+Find Imath include dirs and libraries
+
+Use this module by invoking find_package with the form::
+
+  find_package(Imath
+    [version] [EXACT]      # Minimum or EXACT version
+    [REQUIRED]             # Fail with error if Imath is not found
+    [COMPONENTS <libs>...] # Imath libraries by their canonical name
+                           # e.g. "Half" for "libHalf"
+    )
+
+IMPORTED Targets
+^^^^^^^^^^^^^^^^
+
+``Imath::Imath``
+  The Imath library target.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This will define the following variables:
+
+``Imath_FOUND``
+  True if the system has the Imath library.
+``Imath_VERSION``
+  The version of the Imath library which was found.
+``Imath_INCLUDE_DIRS``
+  Include directories needed to use Imath.
+``Imath_RELEASE_LIBRARIES``
+  Libraries needed to link to the release version of Imath.
+``Imath_RELEASE_LIBRARY_DIRS``
+  Imath release library directories.
+``Imath_DEBUG_LIBRARIES``
+  Libraries needed to link to the debug version of Imath.
+``Imath_DEBUG_LIBRARY_DIRS``
+  Imath debug library directories.
+
+Cache Variables
+^^^^^^^^^^^^^^^
+
+The following cache variables may also be set:
+
+``Imath_INCLUDE_DIR``
+  The directory containing ``Imath/config-auto.h``.
+``Imath_LIBRARY``
+  Individual component libraries for Imath. may include target_link_libraries() debug/optimized keywords.
+``Imath_LIBRARY_RELEASE``
+  Individual component libraries for Imath release
+``Imath_LIBRARY_DEBUG``
+  Individual component libraries for Imath debug
+
+Hints
+^^^^^
+
+Instead of explicitly setting the cache variables, the following variables
+may be provided to tell this module where to look.
+
+``Imath_ROOT``
+  Preferred installation prefix.
+``IMATH_INCLUDEDIR``
+  Preferred include directory e.g. <prefix>/include
+``IMATH_LIBRARYDIR``
+  Preferred library directory e.g. <prefix>/lib
+``IMATH_DEBUG_SUFFIX``
+  Suffix of the debug version of ilmbase libs. Defaults to "_d".
+``SYSTEM_LIBRARY_PATHS``
+  Global list of library paths intended to be searched by and find_xxx call
+``IMATH_USE_STATIC_LIBS``
+  Only search for static ilmbase libraries
+``DISABLE_CMAKE_SEARCH_PATHS``
+  Disable CMakes default search paths for find_xxx calls in this module
+
+#]=======================================================================]
+
+cmake_minimum_required(VERSION 3.12)
+include(GNUInstallDirs)
+
+set(IMATH_REPO "https://github.com/AcademySoftwareFoundation/Imath.git" CACHE STRING
+    "Repo for auto-build of Imath")
+set(IMATH_TAG "master" CACHE STRING
+  "Tag for auto-build of Imath (branch, tag, or SHA)")
+
+find_package(Imath CONFIG QUIET)
+
+if(NOT TARGET Imath::Imath AND NOT Imath_FOUND)
```

Comment:
FetchContent is a huge convenience when developing. To address the concern in the main conversation thread about potential difficulties of reaching across a firewall to the internet, a simple fix, to accommodate both, would be to add a top-level option such as USE_LOCAL_IMATH, default ON. Such a defaulted option would non-disruptively preserve the existing functionality.

Reply:
I think we already have the option you're proposing - there's `USE_IMATH_HALF` which defaults to OFF so as to use the embedded half. When set to ON, it enables the workflow where you can use an externally built Imath.

I don't know if I see the benefit of having three ways to use Imath - (1) using the embedded half, (2) on-demand download through FetchContent and (3) by building against an externally built one. My preference for Half is that we only support (1) and (3) - with (1) remaining the default. Essentially, we're making the assumption that there are two main groups of users - individuals and studios. For the individual user, the embedded half is likely preferable whereas for the firewalled studio the externally built half is likely preferable, so I'm not clear on who would benefit most from the FetchContent workflow?

I should add that I don't necessarily object to introducing FetchContent in general - for dependencies that are too large or cumbersome to embed it might make a lot of sense, but as the Half dependency is so lightweight, I think the embedded option is even better.